### ISSUE TITLE
Unable to install without javac

### ISSUE THREAD

**Issue by laverya** [tngan/samlify#258]
While there seems to be support for using other xml validators (libxml, xmllint), installation still requires javac. It would be greatly appreciated if there was a way to configure the choice of validator before installation - or even better if the default was to something that everyone had installed. 

```
$ export SAML_VALIDATOR=xmllint
$ yarn add samlify@2.4.2
yarn add v1.15.2
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
error /<repo-path>/node_modules/@passify/xsd-schema-validator: Command failed.
Exit code: 1
Command: javac support/XMLValidator.java
Arguments:
Directory: /<repo-path>/node_modules/@passify/xsd-schema-validator
Output:
No Java runtime present, requesting install.
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
```

The makefile suggests https://github.com/tngan/samlify/blob/76130f229e446f7e9f65475c657bae455f2bd720/Makefile#L23 running with `--ignore-scripts`, but while this may work for samlify it breaks quite a few other packages.

In any case, this is preventing us from updating to 2.4.x.

**Comment by tngan** [comment]
@laverya For v2.4.x, we only have javac. SAML_VALIDATOR is starting from v2.5.x.

The makefile said that we need to use `npm install` instead of `yarn add --ignore-scripts` because `--ignore-scripts` would break other packages as you mentioned, I will modify the text to make it clear.

**Comment by laverya** [comment]
Oh, gotcha! It wasn't clear to me that 2.4 only had javac available.

Requiring errors to be manually ignored during installation seems a little odd though - and I'm really not eager to add that to our CI processes. Is there any reason those dependencies aren't being [marked as optional](https://docs.npmjs.com/files/package.json#optionaldependencies)?

**Comment by tngan** [comment]
This change will be released in v2.5 soon.
